src/main/java/org/broadinstitute/hellbender/utils/read/markduplicates/sparkrecords/Pair.java

```diff
@@ -0,0 +1,217 @@
+package org.broadinstitute.hellbender.utils.read.markduplicates.sparkrecords;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import htsjdk.samtools.SAMFileHeader;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.broadinstitute.hellbender.utils.read.markduplicates.MarkDuplicatesScoringStrategy;
+import org.broadinstitute.hellbender.utils.read.markduplicates.OpticalDuplicateFinder;
+import org.broadinstitute.hellbender.utils.read.markduplicates.ReadEnds;
+import org.broadinstitute.hellbender.utils.read.markduplicates.ReadsKey;
+
+/**
+ * Class representing a pair of reads together with accompanying optical duplicate marking information.
+ *
+ * This class holds onto as little information as possible in an attempt to prevent excessive serialization of
+ * during the processing step of MarkDuplicatesSpark
+ */
+@DefaultSerializer(Pair.Serializer.class)
+public final class Pair extends PairedEnds implements OpticalDuplicateFinder.PhysicalLocation {
+    protected transient GATKRead first;
+    protected transient GATKRead second;
```

Comment:
Can you document why these can be transient (as it's not obvious that this is safe).